### DISCUSSION TITLE
Cannot get simple RTSP stream working with gstreamer

### DISCUSSION THREAD

**Discussion by PowercoderJr** [bluenviron/mediamtx#1466] [Questions]
Hi! I'm trying to publish a simple RTSP stream and open it with VLC. The publishing command is:

`gst-launch-1.0 videotestsrc ! queue ! x264enc ! rtspclientsink location=rtsp://localhost:8554/mystream`

It works without errors some time, then gives "Could not read from resource. The server closed the connection." warning and then the timer continues.

![image](https://user-images.githubusercontent.com/25909264/218320637-b38d6406-2a5b-4d14-b564-b917360c4156.png)

On the server side it looks like that (first 3 lines are printed before the warning, last 3 - after):

![image](https://user-images.githubusercontent.com/25909264/218320853-fa337b7b-7f49-47a7-95f5-71d5e2923c75.png)

When I try to open it in VLC (open network stream... -> rtsp://localhost:8554/mystream/) I get additional prints:

![image](https://user-images.githubusercontent.com/25909264/218321644-544d04a0-65d9-44e6-a47b-18974eb81a29.png)

But no stream visible. VLC says "Your input can't be opened: VLC is unable to open the MRL 'rtsp://localhost:8554/mystream/'."

**Comment by aler9** [comment]
Hello, the log is clear about the issue:

```
no UDP packets received in a while
```

make sure that UDP packets can reach the server.

This is usually caused by the presence of a firewall between the client and the server, but since in your case they are both on localhost, the issue is probably related to the fact that your PC resolves `localhost` as a IPv6 address and sends packets to the local IPv6 address, while the server is listening on a IPv4 address. Change `localhost` into `127.0.0.1`. BTW it's a GStreamer issue, not a server one.

**Comment by PowercoderJr** [comment]
Dosen't help, but with TCP option it works fine, so doesn't matter

[a comment by an automated account is not shown]
